src/components/ProgressPage.tsx

```diff
@@ -0,0 +1,72 @@
+import { FC, useEffect, useState } from "react";
+import { Box, styled, Stepper, Typography, Step, StepLabel, StepConnector, StepConnectorProps} from "@mui/material";
+import CheckCircleIcon from '@mui/icons-material/CheckCircle';
+import RadioButtonUncheckedIcon from '@mui/icons-material/RadioButtonUnchecked';
+
+const MemoryTestsLabels = [
+    {
+        "name": "BoCA - Memory (Immediate Recall)",
+        "desc": "BoCA description"
+    },
+    {
+        "name": "Visual Paired Associates",
+        "desc": "VPA desc"
+    },
+    {
+        "name": "Choice Reaction Time",
+        "desc": "CRT desc"
+    },
+    {
+        "name": "Digit Symbol Matching",
+        "desc": "DSM desc"
+    },
+    {
+        "name": "NeurCare - Spatial Memory",
+        "desc": "Spatial Memory desc"
+    },
+    {
+        "name": "BoCA - Memory Recall (Delayed Recall)",
+        "desc": "BoCA delayed recall desc"
+    }
+];
+
+
+interface ProgressPageProps  {  
+    id:number
+}
+
+interface CustomConnectorProps extends StepConnectorProps {
+    numsteps: number;
+  }
+
+const CustomConnector = styled(StepConnector)<CustomConnectorProps>(({ numsteps }) => ({
+    '& .MuiStepConnector-line': {
+      minHeight: `calc(100vh / ${numsteps} - 20px)`,  // 72px accounts approximately for the height of the step button and label
+    },
+  }));
+
+export const ProgressPage: FC<ProgressPageProps> = (props) => {
+    const [activeStep, setActiveStep] = useState(0);
+
+    useEffect(()=> {
+        setActiveStep(props.id)
+    }, [])
+
+    return (
+        <Box alignContent={'center'} sx={{width: '100%', height: '100vh', overflow:'auto', padding: '5vh', display:'flex', flexDirection:'column', alignItems:'center'}}>
+            <Stepper activeStep={activeStep} orientation="vertical" connector={<CustomConnector numsteps={MemoryTestsLabels.length} sx={{paddingLeft:0.5}}/>}>
+               {
+                MemoryTestsLabels.map((step, index) => (
+                    <Step key={index}>
+                        <StepLabel icon={activeStep > index ? <CheckCircleIcon fontSize="large"/> : <RadioButtonUncheckedIcon fontSize="large" color={activeStep === index? "inherit" : "disabled"}/>} sx={{color:activeStep <= index ? '#ffffff' : '#009933'}}>
+                            <Typography sx={{fontSize: 24}} color={activeStep <= index ? activeStep === index ? "#ffffff" : "#808080" : "#009933"}>
```

Comment:
The next step in the stepper is invisible, but it becomes much brighter in dark mode. Probably use another way to show or hide next step?